DOL.WHD.Section14c.Web/src/modules/components/attachmentField/attachmentFieldController.test.js

```diff
@@ -29,34 +29,62 @@ describe('attachmentFieldController', function() {
     })
   );
 
-  it('attachment selected no files, file value should stay the same', function() {
+  it('should only allows specific file types: pdf jpg jpeg png csv PDF', function() {
+    var controller = attachmentFieldController();
+    var fileInput = { files: [{name: 'name1.docx'}] };
+    controller.onAttachmentSelected(fileInput);
+    expect(controller.upload.status).toBe('Invalid');
+    expect(controller.upload.message).toBe('Invalid File Type.');
+    expect(mockApiService.uploadAttachment).not.toHaveBeenCalled();
+  });
```

Comment:
Can you create a set of tests for every valid input as well?  That can be in a loop, too, perhaps something like this (I copied and pasted bits from elsewhere, so this code may not actually work):

```js
['pdf', 'jpg', 'jpeg', 'png', 'csv'].forEach(function(extension) {
  it('should allow files with type ' + extension, function() {
    var controller = attachmentFieldController();
    var fileInput = { files: [{ name: 'file.' + extension, size: 1000 }] };

    controller.onAttachmentSelected(fileInput);
    uploadAttachment.resolve({ data: [{ id: 1, originalFileName: 'name1.pdf' }] });
    scope.$apply();

    expect(scope.attachmentName).toBe('name1.pdf');
    expect(fileInput.value).toBe('');
    expect(controller.upload.status).toBe('Success');
  });
});
```